src/Compilers/Test/Utilities/VisualBasic/ParserTestUtilities.vb

```diff
@@ -10,8 +10,10 @@ Imports Microsoft.CodeAnalysis.Text
 Imports Microsoft.CodeAnalysis.VisualBasic.SyntaxFacts
 Imports Roslyn.Test.Utilities
 Imports Xunit
+Imports Microsoft.CodeAnalysis.Collections
 
 Friend Module ParserTestUtilities
+    Friend ReadOnly Property PooledStringBuilderPool As ObjectPool(Of PooledStringBuilder) = PooledStringBuilder.CreatePool()
```

Comment:
It looks like by default the pool will be limited to only to 32 objects. Perhaps if you increase the size of the single pool to 64, you might get different numbers. Going to a 100 might even give additional improvement. I think it is worth exploring this.